### PR TITLE
add edgecase handling for dashes in aws_domain

### LINKED ISSUE #2: Problem if code_artifact_domain contains dashes
## Summary

If a dash is contained in the user's aws domain name, partifact has trouble parsing the url correctly.

## Steps to Reproduce
Put this in `pyproject.toml` file:
```
[[tool.poetry.source]]
name = "aws"
url = "https://my-domain-123456789012.d.codeartifact.us-west-2.amazonaws.com/pypi/my-repo/simple/"
default = true
```

Then run this command:
partifact login aws
```
> botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the GetAuthorizationToken operation: 2 validation errors detected: Value 'domain-123456789012' at 'domainOwner' failed to satisfy constraint: Member must have length less than or equal to 12; Value 'domain-123456789012' at 'domainOwner' failed to satisfy constraint: Member must satisfy regular expression pattern: [0-9]{12}
```

In this scenario, the rules that are parsing the source url are parsing the `code_artifact_domain` as `my` (instead of `my-domain`), and the `aws_account` as `domain-123456789012` (instead of `123456789012`)

## Expected behavior:

The following url would be parsed as such:
```
url = "https://my-domain-123456789012.d.codeartifact.us-west-2.amazonaws.com/pypi/my-repo/simple/"

code_artifact_domain = my-domain
aws_account = 123456789012
code_artifact_repository = my-repo
```

## Actual Behavior:

The following url is parsed as such:
```
url = "https://my-domain-123456789012.d.codeartifact.us-west-2.amazonaws.com/pypi/my-repo/simple/"

code_artifact_domain = my
aws_account = domain-123456789012
code_artifact_repository = my-repo
```

### PR DESCRIPTION
**Pull request by timothestes**
Resolves https://github.com/Validus-Risk-Management/partifact/issues/2

### CODE REVIEW

File: partifact/config.py [hunk 1]
```diff
@@ -9,6 +10,12 @@
 
 CONFIG_PATH = "./pyproject.toml"
 URL_TEMPLATE = "https://{code_artifact_domain}-{aws_account}.d.codeartifact.{aws_region}.amazonaws.com/pypi/{code_artifact_repository}"
+# Regular expression pattern to parse the URL if a dash is detected in the normally parsed aws_account.
+# (?P<code_artifact_domain>(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)): matches the code_artifact_domain, allowing alphanumeric characters and dashes
+# (?P<aws_account>\d+): matches the aws_account, allowing one or more digits
+# (?P<aws_region>[a-z0-9-]+): matches the aws_region, allowing lowercase alphanumeric characters and dashes
+# (?P<code_artifact_repository>.*): matches the code_artifact_repository, allowing any character (except a newline)
+URL_PATTERN = r"https://(?P<code_artifact_domain>(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?))-(?P<aws_account>\d+).d.codeartifact.(?P<aws_region>[a-z0-9-]+).amazonaws.com/pypi/(?P<code_artifact_repository>.*)"
```
Comment (davidsteiner) [inline partifact/config.py]: Overall, this looks good - thanks. My preference would be using a single approach (either regex or `parse`) for everything, though.

I think something like
```python
URL_PATTERN = "https://(?P<code_artifact_domain>.*)-(?P<aws_account>\d+).d.codeartifact.(?P<aws_region>[a-z0-9-]+).amazonaws.com/pypi/(?P<code_artifact_repository>.*)"
```
would work for both cases.

Alternatively, we could stick with `parse` and go with a template like
```python
URL_TEMPLATE = "https://{subdomain}.d.codeartifact.{aws_region}.amazonaws.com/pypi/{code_artifact_repository}"
```
and `rsplit` the result.

What do you think?

Comment (timothestes) [reply partifact/config.py]: @davidsteiner I think the regex approach would be cleaner and would be my choice moving foward.

I'll update the PR and I'll run the test suite to make sure the newly proposed regex parsing doesn't fail us!

**Comment by timothestes** [comment]
@davidsteiner Let me know what you think about these proposed changes.

Notes:
- If you decided to go this route, you can remove the `parse` library from the list of dependencies, since no longer being used
- `URL_TEMPLATE` variable still being used to create config fixtures in the unit tests. Not sure about how you want to proceed here, so I left the unit test conf alone.

**Review by davidsteiner — APPROVED** [review]

File: partifact/config.py (hunk 1 above)
Comment (davidsteiner) [reply partifact/config.py]: Yeah, I like the regex approach. Thanks!

**Pull request merged**